src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.CreateToken.cs

```diff
@@ -706,11 +706,20 @@ int sizeOfEncodedHeaderAndPayloadAsciiBytes
 
             writer.WriteStartObject();
 
-            if (!string.IsNullOrEmpty(tokenDescriptor.Audience))
+            // TODO at next major version (8.0) use only Audiences as SecurityTokenDescriptor.Audience will be removed.
+            if (!tokenDescriptor.Audiences.IsNullOrEmpty())
```

Comment:
If we make the change where we don't combine Audience and Audiences, we will silently drop the Audience value.
Perhaps we should write them both.

Reply:
Thanks this saved me some time debugging :)